softioc/asyncio_dispatcher.py

```diff
@@ -32,12 +32,18 @@ def aioJoin(worker=worker, loop=self.loop):
         else:
             self.loop = loop
 
-    def __call__(self, func, *args):
+    def __call__(self, func, completion, func_args=(), completion_args=()):
         async def async_wrapper():
             try:
-                ret = func(*args)
-                if inspect.isawaitable(ret):
-                    await ret
+                if inspect.iscoroutinefunction(func):
+                    await func(*func_args)
+                else:
+                    ret = func(*func_args)
+                    # Handle the case of a synchronous function that returns a
+                    # coroutine, like the lambda for on_update_name does
+                    if inspect.isawaitable(ret):
+                        await ret
```

Comment:
Hmm.  Suspect this would be more simply written as
```python
ret = func(*func_args)
if inspect.isawaitable(ret):
    await ret
```

Reply:
Fixed in a0bce0ce330bb96274d4bbebff179c872165d7e6